http-client-spi/src/main/java/software/amazon/awssdk/http/SdkHttpClient.java

```diff
@@ -32,4 +32,5 @@
      */
     AbortableCallable<SdkHttpFullResponse> prepareRequest(SdkHttpFullRequest request, SdkRequestContext requestContext);
 
+    String clientName();
```

Comment:
We should clearly document this since it has the potential to skew our metrics if people misuse it. Also are there any characters that could screw up parsing of the user agent? Would we need to sanitize.